packages/open-next/src/adapters/routing/matcher.ts

```diff
@@ -191,11 +192,12 @@ export function handleRedirects(
     event.rawPath.endsWith("/") &&
     event.rawPath !== "/"
   ) {
+    const headersLocation = event.url.split('?');
     return {
       type: event.type,
       statusCode: 308,
       headers: {
-        Location: event.url.replace(/\/$/, ""),
+        Location: `${headersLocation[0].replace(/\/$/, "")}${headersLocation[1] ? `?${headersLocation[1]}` : ''}`,
```

Comment:
Should the internalEvent.url in the final return below also split the `?`?

Reply:
I code-reviewed that part, and it is already handled by handleRewrites fn.
On handleRewrite, we are doing it like this:
```
const urlQueryString = new URLSearchParams(convertQuery(query)).toString();
const queryString = urlQueryString ? `?${urlQueryString}` : "";
let rewrittenUrl = rawPath;
url: `${rewrittenUrl}${queryString}`,
```

This should be working as expected because we do not add or remove any slash. On my changes, I've gone for split('?') because there are fewer lines of code. But happy to match the style. I'm not sure if there are some code style guidelines somewhere :D

Reply:
Should be fine, doesn't have to match style.